apiServices/fetchDiet/index.js

```diff
@@ -0,0 +1,28 @@
+const url = "https://api.api-ninjas.com/v1/nutrition?query=";
+const apiKey = process.env.NEXT_PUBLIC_EXERCISES_APIKEY;
+
```

Comment:
This way your API key is stored on the client and will be visible to everyone.
It's better to create your own API route and route the results of the nutrition API through that.
Then your API key is not public and you can control access to it via your own API route.

Reply:
thanks! this is great help and a big blunder.